resources/voting-stats.js

```diff
@@ -0,0 +1,112 @@
+        $('#id_no_votes_error').prop('hidden', false);
+    } else {
+        $('#id_no_votes_footer').prop('hidden', true);
+        $('#id_vote_chart').prop('hidden', false);
+        $('#id_has_votes_footer').prop('hidden', false);
+        $('#id_no_votes_error').prop('hidden', true);
+
+        data.sort(function(a, b){return a - b});
+        // Give the graph some padding on both sides.
+        let min_points = Math.max(data[0] - 2, min_possible_vote);
+        let max_points = Math.min(data[data.length - 1] + 2, max_possible_vote);
+
+        let xlabels = [];
+        let voteFreq = [];
+        for (let i = min_points; i <= max_points; i++) {
+            xlabels.push(i);
+            voteFreq.push(0);
+        }
+
+        let max_number_of_votes = 0;
+        let total_votes = 0;
+        let sum = 0;
+        for (let i = 0; i < data.length; i++) {
+            // Assume the data is valid.
+            voteFreq[data[i] - min_points]++;
+            max_number_of_votes = Math.max(max_number_of_votes, voteFreq[data[i] - min_points]);
+            sum += data[i];
+            total_votes++;
+        }
+
+        let mean = sum / total_votes;
+        let half = Math.floor(total_votes / 2);
+        let median = data[half];
+        if (total_votes % 2 === 0) {
+            median = (median + data[half - 1]) / 2;
+        }
+
+        $('#id_mean_vote').prop('innerText', mean.toFixed(2));
+        $('#id_median_vote').prop('innerText', median.toFixed(2));
+        $('#id_num_of_votes').prop('innerText', total_votes);
+
+        const voteData = {
+            labels: xlabels,
+            datasets: [{
+                label: 'Number of votes for this point value',
+                data: voteFreq,
+                borderColor: 'red',
+                backgroundColor: 'pink',
+            }]
+        };
+        const voteDataConfig = {
+            type: 'bar',
+            data: voteData,
+            options: {
+                responsive: true,
+                scales: {
+                    yAxes: [{
+                        ticks: {
+                            precision: 0,
+                            suggestedMax: Math.ceil(max_number_of_votes * 1.2),
+                            beginAtZero: true,
+                        }
+                    }],
+                    xAxes: [{
+                        ticks: {
+                            beginAtZero: false,
+                        }
+                    }]
+                }
+            }
+        };
+        voteChart = new Chart($('#id_vote_chart').get(0), voteDataConfig);
+    }
+}
+
+function updateUserVote(prev_voted_points, voted_points) {
+    let index = data.indexOf(prev_voted_points);
+    if (index > -1) {
+        data.splice(index, 1);
+    }
+    data.push(voted_points);
+}
+
+function deleteUserVote(prev_voted_points) {
+    data.splice(data.indexOf(prev_voted_points), 1);
+}
+
+function startVotingStats(){
```

Comment:
```suggestion
function startVotingStats() {
```

Reply:
Ok I've also checked the JS files, should be fine now...